[
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "",
        "line": 1,
        "column": 0,
        "endLine": null,
        "endColumn": null,
        "path": "contactlog.py",
        "symbol": "too-many-lines",
        "message": "Too many lines in module (7259/1000)",
        "message-id": "C0302"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "",
        "line": 1,
        "column": 0,
        "endLine": null,
        "endColumn": null,
        "path": "contactlog.py",
        "symbol": "missing-module-docstring",
        "message": "Missing module docstring",
        "message-id": "C0114"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "",
        "line": 357,
        "column": 0,
        "endLine": null,
        "endColumn": null,
        "path": "contactlog.py",
        "symbol": "implicit-str-concat",
        "message": "Implicit string concatenation found in call",
        "message-id": "W1404"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "create_connection",
        "line": 512,
        "column": 4,
        "endLine": 512,
        "endColumn": 8,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'conn' from outer scope (line 525)",
        "message-id": "W0621"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "create_connection",
        "line": 516,
        "column": 4,
        "endLine": 517,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'e' from outer scope (line 56)",
        "message-id": "W0621"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "create_connection",
        "line": 516,
        "column": 4,
        "endLine": 517,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"e\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "create_table",
        "line": 528,
        "column": 17,
        "endLine": 528,
        "endColumn": 21,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'conn' from outer scope (line 525)",
        "message-id": "W0621"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "create_table",
        "line": 538,
        "column": 4,
        "endLine": 539,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'e' from outer scope (line 56)",
        "message-id": "W0621"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "create_table",
        "line": 536,
        "column": 8,
        "endLine": 536,
        "endColumn": 9,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"c\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "create_table",
        "line": 538,
        "column": 4,
        "endLine": 539,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"e\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "createTables",
        "line": 543,
        "column": 0,
        "endLine": 543,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Function name \"createTables\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "createTables",
        "line": 719,
        "column": 4,
        "endLine": 719,
        "endColumn": 8,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'conn' from outer scope (line 525)",
        "message-id": "W0621"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "createTables",
        "line": 723,
        "column": 8,
        "endLine": 724,
        "endColumn": 20,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'e' from outer scope (line 56)",
        "message-id": "W0621"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "createTables",
        "line": 723,
        "column": 8,
        "endLine": 724,
        "endColumn": 20,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"e\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "createTables",
        "line": 729,
        "column": 8,
        "endLine": 730,
        "endColumn": 20,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"e\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "createTables",
        "line": 735,
        "column": 8,
        "endLine": 736,
        "endColumn": 20,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"e\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "createTables",
        "line": 741,
        "column": 8,
        "endLine": 742,
        "endColumn": 20,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"e\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "createTables",
        "line": 747,
        "column": 8,
        "endLine": 748,
        "endColumn": 20,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"e\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "warningmessage",
        "line": 765,
        "column": 4,
        "endLine": 765,
        "endColumn": 12,
        "path": "contactlog.py",
        "symbol": "global-variable-undefined",
        "message": "Global variable 'i' undefined at the module level",
        "message-id": "W0601"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "warningmessage",
        "line": 768,
        "column": 4,
        "endLine": 768,
        "endColumn": 6,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"tb\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "warningmessage",
        "line": 777,
        "column": 9,
        "endLine": 777,
        "endColumn": 28,
        "path": "contactlog.py",
        "symbol": "unspecified-encoding",
        "message": "Using open without explicitly specifying an encoding",
        "message-id": "W1514"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "warningmessage",
        "line": 778,
        "column": 34,
        "endLine": 778,
        "endColumn": 35,
        "path": "contactlog.py",
        "symbol": "undefined-loop-variable",
        "message": "Using possibly undefined loop variable 'i'",
        "message-id": "W0631"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "<lambda>",
        "line": 795,
        "column": 23,
        "endLine": 795,
        "endColumn": 51,
        "path": "contactlog.py",
        "symbol": "unnecessary-lambda",
        "message": "Lambda may not be necessary",
        "message-id": "W0108"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 853,
        "column": 12,
        "endLine": 853,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'date' from outer scope (line 820)",
        "message-id": "W0621"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 864,
        "column": 12,
        "endLine": 864,
        "endColumn": 19,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'tmppath' from outer scope (line 51)",
        "message-id": "W0621"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "save",
        "line": 854,
        "column": 12,
        "endLine": 854,
        "endColumn": 24,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"guardianName\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "save",
        "line": 855,
        "column": 12,
        "endLine": 855,
        "endColumn": 25,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"contactMethod\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "save",
        "line": 856,
        "column": 12,
        "endLine": 856,
        "endColumn": 23,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"phoneNumber\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "save",
        "line": 857,
        "column": 12,
        "endLine": 857,
        "endColumn": 24,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"emailAddress\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "save",
        "line": 858,
        "column": 12,
        "endLine": 858,
        "endColumn": 27,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"contactResponse\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "save",
        "line": 859,
        "column": 12,
        "endLine": 859,
        "endColumn": 26,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"contactGeneral\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "save",
        "line": 860,
        "column": 12,
        "endLine": 860,
        "endColumn": 27,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"contactSpecific\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "save",
        "line": 861,
        "column": 12,
        "endLine": 861,
        "endColumn": 24,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"contactNotes\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 870,
        "column": 17,
        "endLine": 870,
        "endColumn": 35,
        "path": "contactlog.py",
        "symbol": "unspecified-encoding",
        "message": "Using open without explicitly specifying an encoding",
        "message-id": "W1514"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 847,
        "column": 17,
        "endLine": 847,
        "endColumn": 22,
        "path": "contactlog.py",
        "symbol": "unused-argument",
        "message": "Unused argument 'event'",
        "message-id": "W0613"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "<lambda>",
        "line": 924,
        "column": 33,
        "endLine": 924,
        "endColumn": 52,
        "path": "contactlog.py",
        "symbol": "unnecessary-lambda",
        "message": "Lambda may not be necessary",
        "message-id": "W0108"
    },
    {
        "type": "error",
        "module": "contactlog",
        "obj": "save",
        "line": 1095,
        "column": 8,
        "endLine": 1095,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "function-redefined",
        "message": "function already defined line 847",
        "message-id": "E0102"
    },
    {
        "type": "refactor",
        "module": "contactlog",
        "obj": "save",
        "line": 1095,
        "column": 8,
        "endLine": 1095,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "too-many-locals",
        "message": "Too many local variables (34/15)",
        "message-id": "R0914"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 1101,
        "column": 12,
        "endLine": 1101,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'date' from outer scope (line 820)",
        "message-id": "W0621"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 1127,
        "column": 12,
        "endLine": 1127,
        "endColumn": 19,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'tmppath' from outer scope (line 51)",
        "message-id": "W0621"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 1235,
        "column": 20,
        "endLine": 1235,
        "endColumn": 32,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'writer_setup' from outer scope (line 174)",
        "message-id": "W0621"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 1133,
        "column": 17,
        "endLine": 1133,
        "endColumn": 35,
        "path": "contactlog.py",
        "symbol": "unspecified-encoding",
        "message": "Using open without explicitly specifying an encoding",
        "message-id": "W1514"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 1192,
        "column": 27,
        "endLine": 1192,
        "endColumn": 45,
        "path": "contactlog.py",
        "symbol": "unspecified-encoding",
        "message": "Using open without explicitly specifying an encoding",
        "message-id": "W1514"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 1229,
        "column": 21,
        "endLine": 1234,
        "endColumn": 25,
        "path": "contactlog.py",
        "symbol": "unspecified-encoding",
        "message": "Using open without explicitly specifying an encoding",
        "message-id": "W1514"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save.data_entry",
        "line": 1243,
        "column": 16,
        "endLine": 1243,
        "endColumn": 20,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'conn' from outer scope (line 525)",
        "message-id": "W0621"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "save.data_entry",
        "line": 1244,
        "column": 16,
        "endLine": 1244,
        "endColumn": 17,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"c\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "refactor",
        "module": "contactlog",
        "obj": "save",
        "line": 1192,
        "column": 27,
        "endLine": 1192,
        "endColumn": 45,
        "path": "contactlog.py",
        "symbol": "consider-using-with",
        "message": "Consider using 'with' for resource-allocating operations",
        "message-id": "R1732"
    },
    {
        "type": "refactor",
        "module": "contactlog",
        "obj": "save",
        "line": 1095,
        "column": 8,
        "endLine": 1095,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "too-many-statements",
        "message": "Too many statements (100/50)",
        "message-id": "R0915"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 1095,
        "column": 17,
        "endLine": 1095,
        "endColumn": 22,
        "path": "contactlog.py",
        "symbol": "unused-argument",
        "message": "Unused argument 'event'",
        "message-id": "W0613"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "graph",
        "line": 1342,
        "column": 12,
        "endLine": 1342,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'conn' from outer scope (line 525)",
        "message-id": "W0621"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "graph",
        "line": 2027,
        "column": 12,
        "endLine": 2027,
        "endColumn": 19,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'tmppath' from outer scope (line 51)",
        "message-id": "W0621"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "graph",
        "line": 1349,
        "column": 12,
        "endLine": 1349,
        "endColumn": 14,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"df\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "graph",
        "line": 1351,
        "column": 12,
        "endLine": 1351,
        "endColumn": 14,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"df\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "graph",
        "line": 1352,
        "column": 12,
        "endLine": 1352,
        "endColumn": 14,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"df\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "graph",
        "line": 1354,
        "column": 12,
        "endLine": 1354,
        "endColumn": 14,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"df\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "graph",
        "line": 1355,
        "column": 12,
        "endLine": 1355,
        "endColumn": 14,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"mu\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "refactor",
        "module": "contactlog",
        "obj": "graph",
        "line": 1335,
        "column": 8,
        "endLine": 1335,
        "endColumn": 17,
        "path": "contactlog.py",
        "symbol": "too-many-statements",
        "message": "Too many statements (85/50)",
        "message-id": "R0915"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "graph",
        "line": 1335,
        "column": 18,
        "endLine": 1335,
        "endColumn": 23,
        "path": "contactlog.py",
        "symbol": "unused-argument",
        "message": "Unused argument 'event'",
        "message-id": "W0613"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "<lambda>",
        "line": 2063,
        "column": 33,
        "endLine": 2063,
        "endColumn": 52,
        "path": "contactlog.py",
        "symbol": "unnecessary-lambda",
        "message": "Lambda may not be necessary",
        "message-id": "W0108"
    },
    {
        "type": "error",
        "module": "contactlog",
        "obj": "save",
        "line": 2457,
        "column": 8,
        "endLine": 2457,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "function-redefined",
        "message": "function already defined line 847",
        "message-id": "E0102"
    },
    {
        "type": "refactor",
        "module": "contactlog",
        "obj": "save",
        "line": 2457,
        "column": 8,
        "endLine": 2457,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "too-many-locals",
        "message": "Too many local variables (75/15)",
        "message-id": "R0914"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 2464,
        "column": 12,
        "endLine": 2464,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'date' from outer scope (line 820)",
        "message-id": "W0621"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 2530,
        "column": 12,
        "endLine": 2530,
        "endColumn": 19,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'tmppath' from outer scope (line 51)",
        "message-id": "W0621"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 2758,
        "column": 20,
        "endLine": 2758,
        "endColumn": 32,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'writer_setup' from outer scope (line 174)",
        "message-id": "W0621"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 2536,
        "column": 17,
        "endLine": 2536,
        "endColumn": 35,
        "path": "contactlog.py",
        "symbol": "unspecified-encoding",
        "message": "Using open without explicitly specifying an encoding",
        "message-id": "W1514"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 2674,
        "column": 27,
        "endLine": 2674,
        "endColumn": 45,
        "path": "contactlog.py",
        "symbol": "unspecified-encoding",
        "message": "Using open without explicitly specifying an encoding",
        "message-id": "W1514"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 2690,
        "column": 21,
        "endLine": 2690,
        "endColumn": 51,
        "path": "contactlog.py",
        "symbol": "unspecified-encoding",
        "message": "Using open without explicitly specifying an encoding",
        "message-id": "W1514"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 2769,
        "column": 21,
        "endLine": 2769,
        "endColumn": 50,
        "path": "contactlog.py",
        "symbol": "unspecified-encoding",
        "message": "Using open without explicitly specifying an encoding",
        "message-id": "W1514"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save.data_entry",
        "line": 2778,
        "column": 16,
        "endLine": 2778,
        "endColumn": 20,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'conn' from outer scope (line 525)",
        "message-id": "W0621"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "save.data_entry",
        "line": 2779,
        "column": 16,
        "endLine": 2779,
        "endColumn": 17,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"c\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "refactor",
        "module": "contactlog",
        "obj": "save",
        "line": 2674,
        "column": 27,
        "endLine": 2674,
        "endColumn": 45,
        "path": "contactlog.py",
        "symbol": "consider-using-with",
        "message": "Consider using 'with' for resource-allocating operations",
        "message-id": "R1732"
    },
    {
        "type": "refactor",
        "module": "contactlog",
        "obj": "save",
        "line": 2457,
        "column": 8,
        "endLine": 2457,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "too-many-statements",
        "message": "Too many statements (227/50)",
        "message-id": "R0915"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 2457,
        "column": 17,
        "endLine": 2457,
        "endColumn": 22,
        "path": "contactlog.py",
        "symbol": "unused-argument",
        "message": "Unused argument 'event'",
        "message-id": "W0613"
    },
    {
        "type": "error",
        "module": "contactlog",
        "obj": "graph",
        "line": 2989,
        "column": 8,
        "endLine": 2989,
        "endColumn": 17,
        "path": "contactlog.py",
        "symbol": "function-redefined",
        "message": "function already defined line 1335",
        "message-id": "E0102"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "graph",
        "line": 2996,
        "column": 12,
        "endLine": 2996,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'conn' from outer scope (line 525)",
        "message-id": "W0621"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "graph",
        "line": 4201,
        "column": 12,
        "endLine": 4201,
        "endColumn": 19,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'tmppath' from outer scope (line 51)",
        "message-id": "W0621"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "graph",
        "line": 3003,
        "column": 12,
        "endLine": 3003,
        "endColumn": 14,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"df\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "graph",
        "line": 3005,
        "column": 12,
        "endLine": 3005,
        "endColumn": 14,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"df\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "graph",
        "line": 3006,
        "column": 12,
        "endLine": 3006,
        "endColumn": 14,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"df\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "graph",
        "line": 3008,
        "column": 12,
        "endLine": 3008,
        "endColumn": 14,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"df\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "graph",
        "line": 3009,
        "column": 12,
        "endLine": 3009,
        "endColumn": 14,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"mu\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "refactor",
        "module": "contactlog",
        "obj": "graph",
        "line": 4199,
        "column": 23,
        "endLine": 4199,
        "endColumn": 47,
        "path": "contactlog.py",
        "symbol": "use-dict-literal",
        "message": "Consider using '{\"font\": dict(size=10)}' instead of a call to 'dict'.",
        "message-id": "R1735"
    },
    {
        "type": "refactor",
        "module": "contactlog",
        "obj": "graph",
        "line": 4199,
        "column": 33,
        "endLine": 4199,
        "endColumn": 46,
        "path": "contactlog.py",
        "symbol": "use-dict-literal",
        "message": "Consider using '{\"size\": 10}' instead of a call to 'dict'.",
        "message-id": "R1735"
    },
    {
        "type": "refactor",
        "module": "contactlog",
        "obj": "graph",
        "line": 4625,
        "column": 23,
        "endLine": 4625,
        "endColumn": 47,
        "path": "contactlog.py",
        "symbol": "use-dict-literal",
        "message": "Consider using '{\"font\": dict(size=10)}' instead of a call to 'dict'.",
        "message-id": "R1735"
    },
    {
        "type": "refactor",
        "module": "contactlog",
        "obj": "graph",
        "line": 4625,
        "column": 33,
        "endLine": 4625,
        "endColumn": 46,
        "path": "contactlog.py",
        "symbol": "use-dict-literal",
        "message": "Consider using '{\"size\": 10}' instead of a call to 'dict'.",
        "message-id": "R1735"
    },
    {
        "type": "refactor",
        "module": "contactlog",
        "obj": "graph",
        "line": 2989,
        "column": 8,
        "endLine": 2989,
        "endColumn": 17,
        "path": "contactlog.py",
        "symbol": "too-many-statements",
        "message": "Too many statements (167/50)",
        "message-id": "R0915"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "graph",
        "line": 2989,
        "column": 18,
        "endLine": 2989,
        "endColumn": 23,
        "path": "contactlog.py",
        "symbol": "unused-argument",
        "message": "Unused argument 'event'",
        "message-id": "W0613"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "<lambda>",
        "line": 4663,
        "column": 33,
        "endLine": 4663,
        "endColumn": 52,
        "path": "contactlog.py",
        "symbol": "unnecessary-lambda",
        "message": "Lambda may not be necessary",
        "message-id": "W0108"
    },
    {
        "type": "error",
        "module": "contactlog",
        "obj": "save",
        "line": 5533,
        "column": 8,
        "endLine": 5533,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "function-redefined",
        "message": "function already defined line 847",
        "message-id": "E0102"
    },
    {
        "type": "refactor",
        "module": "contactlog",
        "obj": "save",
        "line": 5533,
        "column": 8,
        "endLine": 5533,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "too-many-locals",
        "message": "Too many local variables (38/15)",
        "message-id": "R0914"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 5539,
        "column": 12,
        "endLine": 5539,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'date' from outer scope (line 820)",
        "message-id": "W0621"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 5570,
        "column": 12,
        "endLine": 5570,
        "endColumn": 19,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'tmppath' from outer scope (line 51)",
        "message-id": "W0621"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 5705,
        "column": 20,
        "endLine": 5705,
        "endColumn": 32,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'writer_setup' from outer scope (line 174)",
        "message-id": "W0621"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 5577,
        "column": 17,
        "endLine": 5577,
        "endColumn": 35,
        "path": "contactlog.py",
        "symbol": "unspecified-encoding",
        "message": "Using open without explicitly specifying an encoding",
        "message-id": "W1514"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 5644,
        "column": 27,
        "endLine": 5644,
        "endColumn": 45,
        "path": "contactlog.py",
        "symbol": "unspecified-encoding",
        "message": "Using open without explicitly specifying an encoding",
        "message-id": "W1514"
    },
    {
        "type": "refactor",
        "module": "contactlog",
        "obj": "save",
        "line": 5644,
        "column": 27,
        "endLine": 5644,
        "endColumn": 45,
        "path": "contactlog.py",
        "symbol": "consider-using-with",
        "message": "Consider using 'with' for resource-allocating operations",
        "message-id": "R1732"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 5657,
        "column": 27,
        "endLine": 5657,
        "endColumn": 45,
        "path": "contactlog.py",
        "symbol": "unspecified-encoding",
        "message": "Using open without explicitly specifying an encoding",
        "message-id": "W1514"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 5704,
        "column": 21,
        "endLine": 5704,
        "endColumn": 51,
        "path": "contactlog.py",
        "symbol": "unspecified-encoding",
        "message": "Using open without explicitly specifying an encoding",
        "message-id": "W1514"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save.data_entry",
        "line": 5713,
        "column": 16,
        "endLine": 5713,
        "endColumn": 20,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'conn' from outer scope (line 525)",
        "message-id": "W0621"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "save.data_entry",
        "line": 5714,
        "column": 16,
        "endLine": 5714,
        "endColumn": 17,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"c\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "refactor",
        "module": "contactlog",
        "obj": "save",
        "line": 5657,
        "column": 27,
        "endLine": 5657,
        "endColumn": 45,
        "path": "contactlog.py",
        "symbol": "consider-using-with",
        "message": "Consider using 'with' for resource-allocating operations",
        "message-id": "R1732"
    },
    {
        "type": "refactor",
        "module": "contactlog",
        "obj": "save",
        "line": 5533,
        "column": 8,
        "endLine": 5533,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "too-many-statements",
        "message": "Too many statements (118/50)",
        "message-id": "R0915"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "save",
        "line": 5533,
        "column": 17,
        "endLine": 5533,
        "endColumn": 22,
        "path": "contactlog.py",
        "symbol": "unused-argument",
        "message": "Unused argument 'event'",
        "message-id": "W0613"
    },
    {
        "type": "error",
        "module": "contactlog",
        "obj": "graph",
        "line": 5818,
        "column": 8,
        "endLine": 5818,
        "endColumn": 17,
        "path": "contactlog.py",
        "symbol": "function-redefined",
        "message": "function already defined line 1335",
        "message-id": "E0102"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "graph",
        "line": 5825,
        "column": 12,
        "endLine": 5825,
        "endColumn": 16,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'conn' from outer scope (line 525)",
        "message-id": "W0621"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "graph",
        "line": 6566,
        "column": 12,
        "endLine": 6566,
        "endColumn": 19,
        "path": "contactlog.py",
        "symbol": "redefined-outer-name",
        "message": "Redefining name 'tmppath' from outer scope (line 51)",
        "message-id": "W0621"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "graph",
        "line": 5832,
        "column": 12,
        "endLine": 5832,
        "endColumn": 14,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"df\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "graph",
        "line": 5834,
        "column": 12,
        "endLine": 5834,
        "endColumn": 14,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"df\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "graph",
        "line": 5835,
        "column": 12,
        "endLine": 5835,
        "endColumn": 14,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"df\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "graph",
        "line": 5837,
        "column": 12,
        "endLine": 5837,
        "endColumn": 14,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"df\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "convention",
        "module": "contactlog",
        "obj": "graph",
        "line": 5838,
        "column": 12,
        "endLine": 5838,
        "endColumn": 14,
        "path": "contactlog.py",
        "symbol": "invalid-name",
        "message": "Variable name \"mu\" doesn't conform to snake_case naming style",
        "message-id": "C0103"
    },
    {
        "type": "refactor",
        "module": "contactlog",
        "obj": "graph",
        "line": 5818,
        "column": 8,
        "endLine": 5818,
        "endColumn": 17,
        "path": "contactlog.py",
        "symbol": "too-many-statements",
        "message": "Too many statements (90/50)",
        "message-id": "R0915"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "graph",
        "line": 5818,
        "column": 18,
        "endLine": 5818,
        "endColumn": 23,
        "path": "contactlog.py",
        "symbol": "unused-argument",
        "message": "Unused argument 'event'",
        "message-id": "W0613"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "<lambda>",
        "line": 6607,
        "column": 33,
        "endLine": 6607,
        "endColumn": 52,
        "path": "contactlog.py",
        "symbol": "unnecessary-lambda",
        "message": "Lambda may not be necessary",
        "message-id": "W0108"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "",
        "line": 7237,
        "column": 0,
        "endLine": 7240,
        "endColumn": 3,
        "path": "contactlog.py",
        "symbol": "pointless-string-statement",
        "message": "String statement has no effect",
        "message-id": "W0105"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "",
        "line": 7255,
        "column": 17,
        "endLine": 7255,
        "endColumn": 24,
        "path": "contactlog.py",
        "symbol": "undefined-loop-variable",
        "message": "Using possibly undefined loop variable 'monitor'",
        "message-id": "W0631"
    },
    {
        "type": "warning",
        "module": "contactlog",
        "obj": "",
        "line": 7255,
        "column": 32,
        "endLine": 7255,
        "endColumn": 39,
        "path": "contactlog.py",
        "symbol": "undefined-loop-variable",
        "message": "Using possibly undefined loop variable 'monitor'",
        "message-id": "W0631"
    }
]
